ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -1942,10 +1942,10 @@ extension SignalProducerType {
 		// out of scope. This lets us know when we're supposed to dispose the
 		// underlying producer. This is necessary because `struct`s don't have
 		// `deinit`.
-		let token = DeallocationToken()
+		let token = NSObject()
```

Comment:
Using `NSObject` here didn't bother me because of the use of `NSLock` above (although I suppose that will have a counterpart in swift-corelibs-foundation).

I agree though, using `Lifetime` directly makes sense!